File: html/old/backup/sacor2_2008/palindrome.pas
{
	SACO Round 2 2008
	Solution to problem 5 - Palindromes
	by Ben Steenhuisen
}
PROGRAM palindrome;

var	x,y,iters,eCode:longint;
	N,tempS:string[255];
	flag:boolean;


BEGIN
	flag:=true;
	write("Enter starting N: "); read(N);
	while (flag) do 
	begin
		tempS:="";
		for x:=length(N) downto 1 do 
		begin
			tempS := tempS + N[x];
		end;
		if (tempS=N) then 
		begin 
			flag := false;
		end
		else begin 
			iters := iters + 1;
			if iters=100 then flag:=false;
			val(n,y,eCode);
			val(tempS,x,eCode);
			str((x+y),N);
	 	end;
	end;
	if not(iters=100) then writeln(iters," ",N) else writeln("DOES NOT TERMINATE");
END.

File: html/old/backup/sacor2_2008/palindrome.java
// SACO Second Round 2008
// Problem: Palindrome
// Author:  Marco Gallotta

import java.io.*;
import java.util.Scanner;
public class palindrome {
  public static void main(String args[]) {
    Scanner in = new Scanner(System.in);
    System.out.print("Enter N: ");
    int N = in.nextInt();
    int i;
    for (i = 0; i <= 100; i++) {
      int temp = N, rev = 0;
      while (temp > 0) {
        rev = rev * 10 + temp % 10;
        temp /= 10;
      }
      if (N == rev)
        break;
      N += rev;
    }
    if (i > 100)
      System.out.println("DOES NOT TERMINATE");
    else
      System.out.println(i + " " + N);
  }
}
